api/app/src/main/kotlin/packit/service/OutpackServerClient.kt

```diff
@@ -75,6 +79,24 @@ class OutpackServerClient(appConfig: AppConfig) : OutpackServer
         return GenericClient.get(constructUrl(url))
     }
 
+    override fun addFileToZip(
+        filename: String,
+        hash: String,
+        zipOutputStream: ZipOutputStream,
+        request: HttpServletRequest,
+        response: HttpServletResponse
```

Comment:
You shouldn't need request and response at all here. Just the output stream is enough. I think this points at the GenericClient interface being off.

Also from a separation of concerns perspective, I don't think `OutpackServerClient` should have any Zip related code. Th interface should be the same as the existing `getFileByHash` method, but with a streaming output instead of a `ByteArray`. All the Zip manipulation code should be contained in `PacketService`.

I would expect the signature to look like:
```kt
override fun getFileByHash(hash: String, output: OutputStream)
```